### PR TITLE
feat(dev): Add redis-cluster to devservices

### PR DESCRIPTION
**Pull request by beezz**
For testing queues and processing store monitoring for backpressure management of ingest consumers, this PR adds rabbitmq and redis-cluster containers to devservices.

They do not start with the rest of the services by default and need to be enabled using environment variables.

```
export SENTRY_DEV_USE_REDIS_CLUSTER=true
export SENTRY_DEV_USE_RABBITMQ=true
```
afterwards, use `sentry devservices {up | down}` as you would normally do.

One can always manage the services manually even without the above mentioned environment variables in place:

```
sentry devservices {up | down} redis-cluster
sentry devservices {up | down} rabbitmq
```

To use RabbitMQ as your Celery tasks broker you'll need to change your settings:
```
BROKER_URL = "amqp://guest:guest@localhost:5672/sentry"
```
(see https://develop.sentry.dev/services/queue/).

Unless changed in your environment by `SENTRY_CONF` or `DJANGO_CONF` environment variables, this setting lives in `~/.sentry/sentry.conf.py`.

To make use of the `redis-cluster` you will need an additional section in the `redis.clusters` option. See https://github.com/getsentry/sentry/pull/49606/files#diff-94dc6dcb2c82ecd1b4fab0edee177a697e821f1340bd6710392b4095880a0e87R60-R74

Unless changed in your environment by `SENTRY_CONF` or `DJANGO_CONF` environment variables, this setting lives in `~/.sentry/config.yml`.

### CODE REVIEW

File: src/sentry/conf/server.py [hunk 1]
```diff
@@ -2260,6 +2260,14 @@ def build_cdc_postgres_init_db_volume(settings):
             "volumes": {"redis": {"bind": "/data"}},
         }
     ),
+    "redis-cluster": lambda settings, options: (
+        {
+            "image": "grokzen/redis-cluster:7.0.10",
+            "ports": {f"700{idx}/tcp": f"700{idx}" for idx in range(6)},
+            "volumes": {"redis-cluster": {"bind": "/redis-data"}},
+            "environment": {"IP": "0.0.0.0"},
```
Comment (beezz) [inline src/sentry/conf/server.py:2268]: see ~ https://github.com/Grokzen/docker-redis-cluster#important-for-mac-users ~ why it's needed.

File: src/sentry/conf/server.py [hunk 2]
```diff
@@ -2260,6 +2260,14 @@ def build_cdc_postgres_init_db_volume(settings):
             "volumes": {"redis": {"bind": "/data"}},
         }
     ),
+    "redis-cluster": lambda settings, options: (
+        {
+            "image": "grokzen/redis-cluster:7.0.10",
```
[the remark this answers is not in the document]
Comment (beezz) [reply src/sentry/conf/server.py]: Fixed to use ghcr.io

**Comment by asottile-sentry** [comment]
can you show the commands and startup in the PR description?

**Comment by beezz** [comment]
> can you show the commands and startup in the PR description?

Updated PR description.

**Review by loewenheim — APPROVED** [review]

File: src/sentry/data/config/config.yml.default [hunk 3]
```diff
@@ -57,6 +57,21 @@ redis.clusters:
       0:
         host: 127.0.0.1
         port: 6379
+  rc-default:
```
Comment (Swatinem) [inline src/sentry/data/config/config.yml.default:60]: is this enough to use it? or does one need to actually put `rc-default` somewhere else to use it?

**Review by Swatinem — APPROVED** [review]

File: src/sentry/data/config/config.yml.default (hunk 3 above)
Comment (beezz) [reply src/sentry/data/config/config.yml.default:60]: This adds another Redis cluster. You need to configure some service to use it.

**Pull request merged**